src/core/server/ui_settings/ui_settings_service.ts

```diff
@@ -0,0 +1,140 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Observable } from 'rxjs';
+import { first } from 'rxjs/operators';
+import { CoreService } from '../../types';
+import { CoreContext } from '../core_context';
+import { Logger } from '../logging';
+
+import { SavedObjectsClientContract, SavedObjectAttribute } from '../saved_objects/types';
+import { HttpServiceSetup } from '../http';
+import { UiSettingsConfigType } from './ui_settings_config';
+import { IUiSettingsClient, UiSettingsClient } from './ui_settings_client';
+import { mapToObject } from '../../utils/';
+
+interface SetupDeps {
+  http: HttpServiceSetup;
+}
+
+/**
+ * UI element type to represent the settings.
+ * @public
+ * */
+export type UiSettingsType = 'json' | 'markdown' | 'number' | 'select' | 'boolean' | 'string';
+
+/**
+ * UiSettings parameters defined by the plugins.
+ * @public
+ * */
+export interface UiSettingsParams {
+  /** title in the UI */
+  name: string;
+  /** default value to fall back to if a user doesn't provide any */
+  value: SavedObjectAttribute;
+  /** description provided to a user in UI */
+  description: string;
+  /** used to group the configured setting in the UI */
+  category: string[];
+  /** a range of valid values */
+  options?: string[];
+  /** text labels for 'select' type UI element */
+  optionLabels?: Record<string, string>;
+  /** a flag indicating whether new value applying requires page reloading */
+  requiresPageReload?: boolean;
+  /** a flag indicating that value cannot be changed */
+  readonly?: boolean;
+  /** defines a type of UI element */
+  type?: UiSettingsType;
+}
+
+/** @internal */
+export interface UiSettingsServiceSetup {
+  /**
+   * Sets the parameters with default values for the uiSettings.
+   * @param values
+   */
+  setDefaults(values: Record<string, UiSettingsParams>): void;
+  /**
+   * Creates uiSettings client with provided *scoped* saved objects client.
+   * @param values
+   */
+  asScopedToClient(savedObjectsClient: SavedObjectsClientContract): IUiSettingsClient;
+}
+
+/** @internal */
+export class UiSettingsService implements CoreService<UiSettingsServiceSetup> {
+  private readonly log: Logger;
+  private readonly config$: Observable<UiSettingsConfigType>;
+  private readonly uiSettingsDefaults = new Map<string, UiSettingsParams>();
+
+  constructor(private readonly coreContext: CoreContext) {
+    this.log = coreContext.logger.get('ui-settings-service');
+    this.config$ = coreContext.configService.atPath<UiSettingsConfigType>('uiSettings');
+  }
+
+  public async setup(deps: SetupDeps): Promise<UiSettingsServiceSetup> {
+    this.log.debug('Setting up ui settings service');
+    const overrides = await this.getOverrides(deps);
+    const { version, buildNum } = this.coreContext.env.packageInfo;
+
+    return {
+      setDefaults: this.setDefaults.bind(this),
+      asScopedToClient: (savedObjectsClient: SavedObjectsClientContract) => {
```

Comment:
this API exposed only to create UiSettingsClient in RequestHandlerContext

Reply:
Since this is an arrow function, you can use an implicit return:

```ts
asScopedToClient: (savedObjectsClient: SavedObjectsClientContract) =>
  new UiSettingsClient({
    type: 'config',
    id: version,
    buildNum,
    savedObjectsClient,
    defaults: mapToObject(this.uiSettingsDefaults),
    overrides,
    log: this.log,
  }),
```